### PR TITLE
implement indirect branch c test

### PR DESCRIPTION
**Pull request by lialan**

### CODE REVIEW

File: SingleSource/UnitTests/2024-indirect-branch.c [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    return a + b;
+}
+
+int main() {
+    // Declare a function pointer and initialize it to point to the add function
+    unsigned (*funcPtr)(unsigned, unsigned) = add;
+    // Now, call the add function indirectly using the function pointer
+    unsigned result = funcPtr(5, 3);
+
+    if (result != 8) {
```
Comment (thealmarty) [inline SingleSource/UnitTests/2024-indirect-branch.c:12]: You can use our builtin to write to stdout now. `__builtin_delendum_write((int)x)`

Comment (lialan) [reply SingleSource/UnitTests/2024-indirect-branch.c:12]: great! how in the test framework to check stdout prints?

Comment (thealmarty) [reply SingleSource/UnitTests/2024-indirect-branch.c:12]: The stdout goes to the `.log` file. So you can check there. There isn't a better test framework support. Maybe we should add that to the script. One can compare the expected result with the content in the `.log` file. If you want to do it the old way, could you add the positive test as well like the other tests? (So add -neg` to this file's name, and add another one that tests equality) Thanks!

Comment (lialan) [reply SingleSource/UnitTests/2024-indirect-branch.c:12]: Guess there is a new work item -- revamp the testing script and use .log file for expected value checking.

I think the old way should be ditched right away -- checking log file is way better. We should discuss about and agree on a test framework for it.

**Comment by hasekawa-takumi** [comment]
Superceded by #19

**Pull request closed without merging**